gateway/cert.go

```diff
@@ -389,7 +390,20 @@ func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *t
 			}
 		}
 
-		newConfig.ClientAuth = domainRequireCert[hello.ServerName]
+		matches := false
+		for key, clientAuth := range domainRequireCert {
+			req := http.Request{Host: hello.ServerName, URL: &url.URL{}}
+			if mux.NewRouter().Host(key).Match(&req, &mux.RouteMatch{}) {
```

Comment:
Checked the library, was not able to find a better way to do such check...

Otherwise looks good, lets add more tests.